ci-operator/step-registry/mirror-images/enable-qe-catalogsource/mirror-images-enable-qe-catalogsource-commands.sh

```diff
@@ -0,0 +1,157 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# private mirror registry host
+# <public_dns>:<port>
+MIRROR_REGISTRY_HOST=`head -n 1 "${SHARED_DIR}/mirror_registry_url"`
+if [ ! -f "${SHARED_DIR}/mirror_registry_url" ]; then
+    echo "File ${SHARED_DIR}/mirror_registry_url does not exist."
+    exit 1
+fi
+echo "MIRROR_REGISTRY_HOST: ${MIRROR_REGISTRY_HOST}"
+
+# the proxy registry port 6001 for quay.io
+MIRROR_PROXY_REGISTRY_QUAY=`echo "${MIRROR_REGISTRY_HOST}" | sed 's/5000/6001/g' `
+echo "MIRROR_PROXY_REGISTRY_QUAY: ${MIRROR_PROXY_REGISTRY_QUAY}"
+
+# the proxy registry port 6002 for redhat.io
+MIRROR_PROXY_REGISTRY=`echo "${MIRROR_REGISTRY_HOST}" | sed 's/5000/6002/g' `
+echo "MIRROR_PROXY_REGISTRY: ${MIRROR_PROXY_REGISTRY}"
+
+function set_proxy () {
+    if test -s "${SHARED_DIR}/proxy-conf.sh" ; then
+        echo "setting the proxy"
+        # cat "${SHARED_DIR}/proxy-conf.sh"
+        echo "source ${SHARED_DIR}/proxy-conf.sh"
+        source "${SHARED_DIR}/proxy-conf.sh"
+    else
+        echo "no proxy setting."
+    fi
+}
+
+# set the registry auths for the cluster
+function set_cluster_auth () {
+    # get the registry configures of the cluster
+    echo "$ oc extract secret/pull-secret -n openshift-config --confirm --to /tmp "
```

Comment:
If want to print out command line on the log, I suggest to introduce a new function:
```
function run_command() {
    local CMD="$1"
    echo "Running Command: ${CMD}"
    eval "${CMD}"
}
```

Reply:
Done, thanks!